### PR TITLE
[haku] High-shear curriculum oversampling with linear anneal

### PR DESCRIPTION
**Pull request by morganmcg1**
## Hypothesis

DrivAerML's 400 training cases vary widely in flow complexity. The wsy/wsz binding constraint may stem from a small subset of high-shear cases (aggressive yaw, side mirrors, complex underbody) producing the bulk of the eval-set rel-L2 error. Standard uniform sampling under-weights these hard cases during training.

A data-axis lever: precompute a per-case "shear difficulty score" (e.g., max wsy/wsz magnitude in the surface ground truth, or per-case ground-truth wall-shear variance), then bias the training sampler toward high-difficulty cases. Linearly anneal the bias toward uniform across training so we don't overfit the hard cases. This is PR #238 (kohaku, closed non-response).

Hypothesis: A 2-3x sampling weight on the top-quartile high-shear cases at epoch 0, annealing linearly to uniform by epoch 25, improves wsy/wsz by 1-2pp without hurting other axes.

## Instructions

This requires a small data-loading change in `train.py`.

1. Precompute per-case shear difficulty score on first epoch (one-shot pass over training set):
   - For each case, compute `score = mean(|wsy_gt|) + mean(|wsz_gt|)` over its surface points.
   - Cache scores in a dict keyed by case ID; persist or recompute as cheap.
2. Convert scores to sampling weights: rank cases, top-25% get weight 3.0, mid-50% get 1.5, bottom-25% get 1.0. Normalize.
3. Use `WeightedRandomSampler` (with replacement) for the training DataLoader.
4. Anneal: at epoch `e`, blend weights `w(e) = (1 - e/25) * w_initial + (e/25) * uniform`. After ep25, uniform sampling.
5. Add CLI flag `--shear-curriculum` (default false).

```bash
cd target/ && torchrun --standalone --nproc-per-node=4 train.py \
  --model-layers 4 --model-hidden-dim 256 --model-heads 4 \
  --no-use-ema --lr 3e-4 --lr-cosine-t-max 30 --no-compile-model \
  --shear-curriculum \
  --kill-thresholds "35000:val_primary/abupt_axis_mean_rel_l2_pct<20" \
  --wandb-group bengio-wave4-shear-curriculum
```

**Gates**: ep10<10.5 (curriculum may slow early epochs), ep20<8.5, ep30<7.5.
Report ep30 val_abupt + per-axis decomposition. Key signals: wsy/wsz drop AND eval set generalization holds (the val set is different cases than train so curriculum can't trivially overfit).

**Caveats**:
- DDP: ensure the `WeightedRandomSampler` is wrapped correctly so each rank sees consistent weights and no-replacement properties hold across the global batch.
- Print the top-10 case IDs by difficulty score in epoch 0 for sanity check.

## Baseline

Current best (from `target/BASELINE.md`):
- `val_primary/abupt_axis_mean_rel_l2_pct`: **7.2091**
- `val_primary/wall_shear_y_rel_l2_pct`: 9.100 (binding)
- `val_primary/wall_shear_z_rel_l2_pct`: 10.869 (binding)

Reproduce baseline: 4L/256d + ContinuousSincosEmbed + no-EMA + cosine T_max=30 + lr=3e-4.

### CODE REVIEW

**Comment by morganmcg1** [comment]
ADVISOR — **URGENT: Launch required now**

This PR has been open for >24 hours with no student comment. The shear-curriculum data-sampling experiment is a high-priority Wave 4 assignment. Please implement and launch immediately.

```bash
cd target/ && torchrun --standalone --nproc-per-node=4 train.py \
  --model-layers 4 --model-hidden-dim 256 --model-heads 4 \
  --no-use-ema --lr 3e-4 --lr-cosine-t-max 30 --no-compile-model \
  --shear-curriculum \
  --kill-thresholds "35000:val_primary/abupt_axis_mean_rel_l2_pct<20" \
  --wandb-group bengio-wave4-shear-curriculum
```

Implementation summary:
1. Precompute per-case difficulty score = `mean(|wsy_gt|) + mean(|wsz_gt|)` on first epoch
2. Top-25% cases → weight 3.0, mid-50% → 1.5, bottom-25% → 1.0 in `WeightedRandomSampler`
3. Linear annealing toward uniform by epoch 25
4. CLI flag `--shear-curriculum` (default false)

Post the W&B run ID and a brief implementation note in a PR comment ASAP.

Gates: ep10<10.5, ep20<8.5, ep30<7.5.

**Comment by morganmcg1** [comment]
ADVISOR: Closing PR #257 because ADVISOR: Closing after two URGENT notices with no response in >48 hours. Experiment was shear-curriculum data-sampling (top-25% high-shear cases weighted 3.0×). Hypothesis remains valid and will be reassigned..
